CMSgov/mint-app
# [NOREF] Add count of linked solutions for documents

# NOREF

## Changes and Description

- Adds `numLinkedSolutions` to the `type PlanDocument` in GraphQL, allowing querying of the number of linked solutions for any given document.

This is useful for displaying information in modals when going to delete or unlink a document.

## How to test this change

- Start up the service locally with `scripts/dev up`
- Seed the database with `scripts/dev db:seed`
- In one of the Model Plans, upload a new plan document and note the `id` of that document
- Answer a question in the IT Solutions tracker that creates an operational need
- Add an operational solution, and note the `id` of that solution
- Use Postman / GQL Playground to create a plan document solution link:
```gql
mutation createPlanDocumentSolutionLinks ($solutionID: UUID!, $documentIDs: [UUID!]!) {
    createPlanDocumentSolutionLinks (solutionID: $solutionID, documentIDs: $documentIDs) {
        id
        solutionID
        documentID
        createdBy
        createdDts
        modifiedBy
        modifiedDts
    }
}
```
- Query the document and see that it has 1 linked solution:
```gql
query planDocument ($id: UUID!) {
    planDocument (id: $id) {
        numLinkedSolutions
    }
}
```

## PR Author Review Checklist



## PR Reviewer Guidelines
- It's best to pull the branch locally and test it, rather than just looking at the code online!
- Check that all code is adequately covered by tests - if it isn't feel free to suggest the addition of tests.
- Always make comments, even if it's minor, or if you don't understand why something was done.
